core/src/util/inputHandler.cpp

```diff
@@ -100,9 +100,15 @@ void InputHandler::handleFlingGesture(float _posX, float _posY, float _velocityX
 void InputHandler::handlePinchGesture(float _posX, float _posY, float _scale, float _velocity) {
     cancelFling();
 
+    // Early out for an invalid scale
+    if (_scale == 0.f) {
+        return;
+    }
     float z = m_view.getZoom();
     static float invLog2 = 1 / log(2);
-    m_view.zoom(log(_scale) * invLog2);
+
+    // Make sure log is valid with absolute value of scale
+    m_view.zoom(log(abs(_scale)) * invLog2);
```

Comment:
If `_scale` is negative, zooming using the absolute value is probably not the expected result. Maybe instead we should exit early when `_scale <= 0`?

Reply:
Probably! But I did not see any adverse side effect of this in my very limiting testing. I have to think this through but I feel it might be fully accurate to simply ignore negative scale values.. Do you have a good reasoning behind ignoring negative scales, if so if you could take over and update with a code comment explaining the same? Thanks